proto/redpanda/api/console/v1alpha1/security.proto

```diff
@@ -0,0 +1,751 @@
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+
+  // Members assigned to the role.
+  repeated RoleMembership added = 2;
+
+  // Members removed from the role.
+  repeated RoleMembership removed = 3;
+}
+
+// ListUserRolesRequest is the request for ListUserRoles.
+message ListUserRolesRequest {
+  // Filter options.
+  message Filter {
+    // Filter results to only roles named with the prefix.
+    string name_prefix = 1 [
+      (buf.validate.field).string.max_len = 128,
+      (buf.validate.field).string.pattern = "^([a-zA-Z0-9-_]*)$"
+    ];
+  }
+
+  // Optional filter.
+  optional Filter filter = 1;
+
+  // Page size.
+  int32 page_size = 2 [
+    (buf.validate.field).int32 = {
+      gte: -1,
+      lte: 1000
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Limit the paginated response to a number of items. Defaults to 100. Use -1 to disable pagination.",
+      minimum: -1,
+      maximum: 1000
+    }
+  ];
+
+  // Value of the next_page_token field returned by the previous response.
+  // If not provided, the system assumes the first page is requested.
+  string page_token = 3;
+}
+
+// ListUserRolesResponse is the response for ListUserRoles.
+message ListUserRolesResponse {
+  // The roles in the system.
+  repeated Role roles = 1;
+}
+
+// ListRolesWithMembersRequest is the request for ListRolesWithMembers.
+message ListRolesWithMembersRequest {
+  // Filter options.
+  message Filter {
+    // Filter results only roles named with the prefix.
+    string name_prefix = 1 [
+      (buf.validate.field).string.max_len = 128,
+      (buf.validate.field).string.pattern = "^([a-zA-Z0-9-_]*)$"
+    ];
+  }
+
+  // Optional filter.
+  optional Filter filter = 1;
+
+  // Page size.
+  int32 page_size = 2 [
+    (buf.validate.field).int32 = {
+      gte: -1,
+      lte: 1000
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Limit the paginated response to a number of items. Defaults to 100. Use -1 to disable pagination.",
+      minimum: -1,
+      maximum: 1000
+    }
+  ];
+
+  // Value of the next_page_token field returned by the previous response.
+  // If not provided, the system assumes the first page is requested.
+  string page_token = 3;
+}
+
+// RoleWithMembers is role data with members assigned to the role.
+message RoleWithMembers {
+  // The name of the role.
+  string name = 1;
+
+  // Members assigned to the role.
+  repeated RoleMembership members = 2;
+}
+
+// ListRolesWithMembersResponse is the response for ListRolesWithMembers.
+message ListRolesWithMembersResponse {
+  // The roles in the system.
+  repeated RoleWithMembers roles = 1;
+}
+
+service SecurityService {
+  // ListRoles lists all the roles based on optional filter.
+  rpc ListRoles(ListRolesRequest) returns (ListRolesResponse) {
+    option (google.api.http) = {get: "/v1alpha1/security/roles"};
```

Comment:
I would remove all HTTP annotations since these endpoints are just for Console only as of today, right? I imagine the `/security` prefix does not align with the existing structure of the endpoints.

Reply:
Removed all `google.http.api` and grpc annotations.